internal/common/conversion/import_state.go

```diff
@@ -0,0 +1,57 @@
+package conversion
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+)
+
+func IDWithProjectIDClusterName(projectID, clusterName string) (string, error) {
+	if err := ValidateProjectID(projectID); err != nil {
+		return "", err
+	}
+	if err := ValidateClusterName(clusterName); err != nil {
+		return "", err
+	}
+	return projectID + "-" + clusterName, nil
+}
+
+func ImportStateProjectIDClusterName(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	parts := strings.Split(req.ID, "-")
+	if len(parts) != 2 {
+		resp.Diagnostics.AddError("invalid import ID", "expected 2 parts with project_id and cluster_name: "+req.ID)
+		return
+	}
+	projectID, clusterName := parts[0], parts[1]
+	if err := ValidateProjectID(projectID); err != nil {
+		resp.Diagnostics.AddError("invalid project_id in import ID", err.Error())
+	}
+	if err := ValidateClusterName(clusterName); err != nil {
+		resp.Diagnostics.AddError("invalid cluster_name in import ID", err.Error())
+	}
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("project_id"), projectID)...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("cluster_name"), clusterName)...)
+}
+
+func ValidateProjectID(projectID string) error {
+	re := regexp.MustCompile("^([a-f0-9]{24})$")
+	if !re.MatchString(projectID) {
+		return fmt.Errorf("project_id must be a 24 character hex string: %s", projectID)
+	}
+	return nil
+}
+
+func ValidateClusterName(clusterName string) error {
```

Comment:
this is the current Atlas cluster name regexp, don't know if the regexp or message are too specific and we should rely on Atlas.  
I feel like it's ok in this case as project_id and cluster_names are broadly used.